### PR TITLE
feat(indexer): Implement a response cache

### PR DESCRIPTION
**Pull request by Yawning**

### CODE REVIEW

**Comment by Yawning** [comment]
I have no idea what's up with `lint-go` because I didn't even touch those files.  So I am going to ignore it.

File: indexer/backend_cache.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+	blockGasLimit uint64,
+) error {
+	return cb.inner.Index(ctx, oasisBlock, txResults, blockGasLimit)
+}
+
+func (cb *cachingBackend) Prune(
+	ctx context.Context,
+	round uint64,
+) error {
+	// Yes, this does not prune entries from the cache.  It would be
```
Comment (ptrus) [inline indexer/backend_cache.go:72]: :+1:

File: indexer/backend_cache.go [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+	ctx context.Context,
+	txHash ethcommon.Hash,
+) (map[string]interface{}, error) {
+	return cb.inner.GetTransactionReceipt(ctx, txHash)
+}
+
+func (cb *cachingBackend) BlockNumber(
+	ctx context.Context,
+) (uint64, error) {
+	// XXX/yawning: I assume that this can just be the last indexed round?
```
Comment (ptrus) [inline indexer/backend_cache.go:245]: :+1:
In the old messy code indexing blocks was not done in a transaction :see_no_evil: that's probably this is separate from  `QueryLastIndexedRound` feel free to open an issue to clean this up.

**Review by ptrus — APPROVED** [review]
Nice! 
Please amend the commit to start with: `feature(indexer): ...` so that the `git-lint` will pass :grin: as commits are used to generate the changelog: https://github.com/oasisprotocol/emerald-web3-gateway/blob/main/CONTRIBUTING.md#git-commit-messages

**Comment by ptrus** [comment]
> I have no idea what's up with lint-go because I didn't even touch those files. So I am going to ignore it.

Yeah no idea, running the `lint-go` locally works though, might be a caching problem, will take a look.

**Pull request merged**